cg/meta/clean/clean_flow_cells.py

```diff
@@ -0,0 +1,178 @@
+            )
+        sample_sheet_path: Path = Path(
+            get_sample_sheets_from_latest_version(
+                flow_cell_id=self.flow_cell.id, hk_api=self.hk_api
+            )[0].full_path
+        )
+        self.flow_cell.set_sample_sheet_path_hk(sample_sheet_path)
+
+    def can_flow_cell_directory_be_deleted(self) -> bool:
+        """Determine whether a flow cell directory can be deleted."""
+        return all(
+            [
+                self.is_directory_older_than_21_days(),
+                self.is_flow_cell_in_statusdb(),
+                self.is_flow_cell_backed_up(),
+                self.has_sequencing_metrics_in_statusdb(),
+                self.has_fastq_files_for_samples_in_housekeeper(),
+                self.has_spring_meta_data_files_for_samples_in_housekeeper(),
+                self.has_spring_files_for_samples_in_housekeeper(),
+                self.has_sample_sheet_in_housekeeper(),
+            ]
+        )
+
+    def is_directory_older_than_21_days(self) -> bool:
+        """Check if a given directory is older than 21 days."""
+        return is_directory_older_than_days_old(
+            directory_path=self.flow_cell.path,
+            days_old=TWENTY_ONE_DAYS,
+            current_time=self.current_time,
+        )
+
+    def is_flow_cell_in_statusdb(self) -> bool:
+        """Check if flow cell is in statusdb."""
+        return bool(self.get_flow_cell_from_status_db())
+
+    def is_flow_cell_backed_up(self) -> bool:
+        """Check if flow cell is backed up on PDC."""
+        return bool(self.get_flow_cell_from_status_db().has_backup)
+
+    def has_sequencing_metrics_in_statusdb(self) -> bool:
+        """Check if a flow cell has entries in the SampleLaneSequencingMetrics table."""
+        return bool(self.get_sequencing_metrics_for_flow_cell())
+
+    def has_sample_sheet_in_housekeeper(self) -> bool:
+        """Check if the flow cell has a sample sheet in housekeeper."""
+        return bool(self.flow_cell.get_sample_sheet_path_hk())
+
+    def has_fastq_files_for_samples_in_housekeeper(self) -> bool:
+        """Check if all samples on the flow cell have fastq files in housekeeper."""
+        return bool(self.has_files_for_samples_on_flow_cell_with_tag(tag=SequencingFileTag.FASTQ))
+
+    def has_spring_files_for_samples_in_housekeeper(self) -> bool:
+        """Check if all samples on the flow cell have SPRING files in housekeeper."""
+        return bool(self.has_files_for_samples_on_flow_cell_with_tag(tag=SequencingFileTag.SPRING))
+
+    def has_spring_meta_data_files_for_samples_in_housekeeper(self) -> bool:
+        """Check if all samples on the flow cell have SPRING metadata files in housekeeper."""
+        return bool(
+            self.has_files_for_samples_on_flow_cell_with_tag(tag=SequencingFileTag.SPRING_METADATA)
+        )
+
+    def get_flow_cell_from_status_db(self) -> Optional[Flowcell]:
```

Comment:
Im thinking this could be a property

Reply:
I don't think I would want this as a property as I don't think the flow cell model should be an attribute of the class since it is a condition that has to be fulfilled. Unlike the `FlowCellDirectoryData` which has to exist for the clean api to run.

Reply:
Fair enough